console/react/src/App.js

```diff
@@ -28,26 +29,22 @@ import "@patternfly/patternfly/components/Nav/nav.css";
 import { QDRService } from "./common/qdrService";
 import "./App.css";
 import PageLayout from "./overview/dashboard/layout";
-class App extends Component {
-  state = {};
-
-  render() {
-    // service is passed in to make testing easier
-    const service = new QDRService();
-    // also, a router is used here to provide PageLayout with a history property
-    return (
-      <Router>
-        <div className="App pf-m-redhat-font">
-          <Route
-            path="/"
-            render={props => (
-              <PageLayout service={service} {...props} config={this.props.config} />
-            )}
-          />
-        </div>
-      </Router>
-    );
-  }
+
+function App(props) {
+
+  // service is passed in to make testing easier
+  let service = new QDRService();
+
+  // also, a router is used here to provide PageLayout with a history property
+  let history = createHashHistory({ window });
+
+  return (
+    <HistoryRouter history={history}>
```

Comment:
Why we are using HistoryRouter here? Before this refactoring, we used the HashRouter.

Reply:
Because the code of some components nested further in needs to have `history` instance in properties. That is not available in react-router-dom v6 unless I put it there. And `HistoryRouter` is the way to inject `history`.

Reply:
Here I see that the child component PageLayout pass explicitly the pro ```history={history}``` and history coming from ``` createHashHistory({ window });```

Reply:
Yeah, that is how you are supposed to use HistoryRouter, according to docs, https://reactrouter.com/docs/en/v6/api#unstable_historyrouter

Honestly, there are two things in the console, related to react-router-dom, that I think can be now done better

- navigating between pages; which is now done inconsistently, either with `<Navigate>`, or `navigate()`, or through `history`.
- the page structure which is supposed to use the `<Outlet>` feature but it does not

Reply:
Good point +1.

Probably one of the things to do is refactor the PageLayout component as a function (but this is out of the scope of this PR honestly) to use directly hooks. So we can avoid passing history and use directly useNavigate and useLocation. Then probably we can avoid passing {...props } because without typescript we lose a bit the control of what we inject in the component. Moreover the High order component 'WithRoute'  will be deleted

Reply:
Noted in https://issues.apache.org/jira/browse/DISPATCH-2331